### PR TITLE
Fix null duration bug

### PR DESCRIPTION
**Pull request by Charlie-Henry**
Discovered a bug with some of the parking data where we get a null for the `duration_min` field. This was causing postgres to not allow us to write those files to the DB. This fix will now calculate the duration of the parking session (end time minus start time) if it is null.

### CODE REVIEW

File: meters/smartfolio_s3.py [hunk 1]
```diff
@@ -216,6 +216,21 @@ def transform(smartfolio):
...
+    smartfolio["start_datetime"] = pd.to_datetime(
+        smartfolio["start_time"], format="%Y-%m-%d %H:%M:%S", infer_datetime_format=True
+    )
+    smartfolio["end_datetime"] = pd.to_datetime(
+        smartfolio["end_time"], format="%Y-%m-%d %H:%M:%S", infer_datetime_format=True
+    )
+    smartfolio["timedelta"] = (
+        smartfolio["end_datetime"] - smartfolio["start_datetime"]
+    ).dt.total_seconds()
+    smartfolio.loc[smartfolio["duration_min"].isna(), "duration_min"] = smartfolio.loc[
```
Comment (mateoclarke) [inline meters/smartfolio_s3.py:232]: I don't think I understand whats happening in this line. Is it only writing `"timedelta"` in fields where `"duration_min"` is not a number?

**Review by mateoclarke — APPROVED** [review]
_As I admit in my comment on line 230, I didn't fully understand the functionality of some of the panda operations so I could be totally off but..._

It looks like this new calculation for duration would effect all records, not just those that are null. Just wanted to note that and confirm it was your intention. I wonder if calculating across the board with this method is more accurate/consistent or if it makes sense to only calculate in cases where the value is null. Since you're most familiar with the data, I totally trust your choice on how to proceed.

File: meters/smartfolio_s3.py (hunk 1 above)
Comment (Charlie-Henry) [reply meters/smartfolio_s3.py:232]: Correct, `df.loc[rows, columns]` is the way I'm using to select the part the of the I dataframe need. 

`smartfolio["duration_min"].isna()` is just a boolean series so this will select only the rows in the dataframe where `duration_min` is null.

I tested this and was getting what I needed but I agree sometimes Pandas can get really wonky with this syntax. There is probably a better way like using a lambda like [here](https://github.com/cityofaustin/atd-parking-data/blob/5e92da5373ff0ab71b87b752d9708db91b0214f7/meters/smartfolio_s3.py#L186) but that can also be pretty hard to follow.

Just realized I didn't divide the time by 60 to convert to minutes...
